src/extension/index.activateGithubAnalyses.ts

```diff
@@ -110,13 +123,23 @@ export function activateGithubAnalyses(disposables: Disposable[], store: Store,
         }
 
         const [, user, repoName] = (() => {
-            // Example: https://github.com/user/repoName.git
             const matchHTTPS = origin.match(/https:\/\/github\.com\/([^/]+)\/([^/]+)/);
             if (matchHTTPS) return matchHTTPS;
 
-            // Example: git@github.com:user/repoName.git
-            const matchSSH = origin.match(/git@github\.com:([^/]+)\/([^/]+)/);
-            if (matchSSH) return matchSSH;
+            // Try githubCodeScanningEndpoint first, then try it without a "api." prefix.
+            // This is to support GitHub Enterprise Server, which may have a different endpoint.
+
+            const endpoints = [gitHubCodeScanningEndpoint, gitHubCodeScanningEndpoint.replace('api.', '')];
+
+            for (const endpoint of endpoints) {
+                // using gitHubCodeScanningEndpoint to match the endpoint used by the GitHub Code Scanning API
+                const matchHTTPS = origin.match(new RegExp(`${endpoint}/([^/]+)/([^/]+)`));
+                if (matchHTTPS) return matchHTTPS;
+
+                // Example: git@github.com:user/repoName.git
+                const matchSSH = origin.match(new RegExp(`git@${endpoint.replace('https://', '')}:([^/]+)/([^/]+)`));
```

Comment:
To prevent this `https://` stripping, can the setting be just the hostname (`api.github.com`) rather than the URL (`https://api.github.com`)?